### ISSUE TITLE
bun build with --sourcemap breaks file imports

### ISSUE THREAD

**Issue by mpartel** [oven-sh/bun#6553]
### What version of Bun is running?

1.0.6+969da088f5db3258a803ec186012e30f992829b4

### What platform is your computer?

Linux 6.5.4-76060504-generic x86_64 x86_64

### What steps can reproduce the bug?

1. `bun init`
2. Put this in `index.ts`:
```typescript
import bla from './bun.lockb';
console.log(bla);
```
3. Run `bun build index.ts --outdir out --sourcemap=external`  (the sourcemap flag triggers the bug)
4. View `out/index.js`

### What is the expected behavior?

The import should be translated to a relative path to the file that bun copied to `out/` ([docs](https://bun.sh/docs/bundler/loaders#file)).

### What do you see instead?

The import was translated into an absolute path to the file on my machine (as [expected](https://bun.sh/docs/bundler/loaders#file) for `bun run` but not `bun build`).

### Additional information

Also broken with `--sourcemap=inline`.

Behaves correctly if sourcemap flag is removed.

**Comment by LukeClancy** [comment]
I ran into the same bug, along with the sourcemap SegFault bug. Seeing some of the other sourcemap bugs around, I think people got the message, so I wont create another issue. However - I did create a testing utility demonstrating the issue, and I have another note.

- The sourcemap segfaults seem to particularly happen on jpegs.

Heres my utility: [https://github.com/LukeClancy/bun_error](https://github.com/LukeClancy/bun_error)
